onnxruntime/core/mlas/lib/sgemm.cpp

```diff
@@ -1380,77 +1407,97 @@ Return Value:
 
 --*/
 {
-    float PanelA[MLAS_SGEMM_TRANSA_ROWS * MLAS_SGEMM_PACKED_STRIDEK];
-
-    //
-    // Step through each slice of matrix B along the N dimension.
-    //
-
-    size_t CountN;
+#if !(defined(USE_KLEIDIAI) && !defined(_MSVC_LANG))
+    MLAS_UNREFERENCED_PARAMETER(TransB);
```

Comment:
Ignoring it for non-Kleidi enabled builds seems like a bad idea in general as the caller might be oblivious to its non-usage.